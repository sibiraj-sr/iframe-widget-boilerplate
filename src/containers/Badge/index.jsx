import React from 'react';
import styled from 'styled-components';

const BadgeIcon = styled.div`
  border: 0.25rem solid #fff;
  width: 72px;
  height: 72px;
  box-shadow: 0 3px 10px 0 rgba(0, 0, 0, 0.1);
`;

const BadgeConent = styled.div`
  width: 11.5rem;
  height: 3rem;
  display: flex;
  align-items: center;
  box-shadow: 0 3px 10px 0 rgba(0, 0, 0, 0.1);
`;

function Badge () {
  return (
    <div className='flex items-center cursor-pointer p-2 animated fadeIn'>
      <BadgeConent className='bg-green-800 py-3 px-6 w-48 -mr-4 rounded-l-lg'>
        <p className='text-white font-medium tracking-wide'>Chat with us!</p>
      </BadgeConent>
      <BadgeIcon className='bg-green-800 border-white rounded-full flex items-center justify-center'>
        <i className='h-6 w-6' data-feather='message-square' fill='#fff' stroke='#fff' />
      </BadgeIcon>
    </div>
  );
}

export default Badge;
