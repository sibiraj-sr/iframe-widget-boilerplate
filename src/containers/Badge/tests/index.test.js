import React from 'react';
import { render } from '@testing-library/react';

import Badge from '../index';

describe('<Badge />', () => {
  it('should render and match the snapshot', () => {
    const { asFragment } = render(<Badge />);
    const renderedOutput = asFragment();
    expect(renderedOutput).toMatchSnapshot();
  });
});
