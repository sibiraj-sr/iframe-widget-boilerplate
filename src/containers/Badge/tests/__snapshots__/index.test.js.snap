// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render and match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center cursor-pointer p-2 animated fadeIn"
  >
    <div
      class="Badge__BadgeConent-sc-12uub00-1 dyIDMi bg-green-800 py-3 px-6 w-48 -mr-4 rounded-l-lg"
    >
      <p
        class="text-white font-medium tracking-wide"
      >
        Chat with us!
      </p>
    </div>
    <div
      class="Badge__BadgeIcon-sc-12uub00-0 VLUKT bg-green-800 border-white rounded-full flex items-center justify-center"
    >
      <i
        class="h-6 w-6"
        data-feather="message-square"
        fill="#fff"
        stroke="#fff"
      />
    </div>
  </div>
</DocumentFragment>
`;
